server/models/meta.py

```diff
@@ -0,0 +1,263 @@
+            obj[key] = obj[key].upper()
+
+    def __lower(self, obj, key):
+        if key in obj:
+            obj[key] = obj[key].lower()
+
+    def _parse(self, name, regex):
+        m = regex.match(name)
+        if m is None:
+            raise ValidationException('Invalid name')
+        d = m.groupdict()
+        self.__lower(d, 'uuid')
+        self.__upper(d, 'barcode')
+        self.__upper(d, 'case')
+        return d
+
+    def parseImage(self, name):
+        """Parse a slide image file name."""
+        return self._parse(name, self.image_re)
+
+    def parsePathology(self, name):
+        """Parse a pathology report file name."""
+        return self._parse(name, self.pathology_re)
+
+    @classmethod
+    def setMimeType(cls, doc):
+        """Set the mime type of a file document."""
+        doc['mimeType'] = mimetypes.guess_type(
+            doc.get('name', '')
+        )[0]
+        return True
+
+    @classmethod
+    def setJobStatus(cls, msg, status=None, logger=None, job=None, **kwargs):
+        """Update a job document with the given message and status."""
+        if job:
+            if status is None:
+                status = job['status']
+            jobModel = ModelImporter.model('job')
+            jobModel.updateJob(
+                job, log=msg,
+                status=status
+            )
+        if logger:
+            logger(msg)
+        return job
+
+    def importDocument(self, doc, **kwargs):
+        """Promote a Girder core document to a TCGA model."""
+        self.setTCGA(doc)
+        self.save(doc)
+        if doc.get('_modelType') == 'item':
+            fileModel = self.model('file')
+            files = self.model('item').childFiles(doc)
+            for file in files:
+                if self.setMimeType(file):
+                    fileModel.save(file)
+        return doc
+
+    def loadDocument(self, id, **kwargs):
+        """Load a user, folder, item, or collection document."""
+        for modelType in ('collection', 'user', 'folder', 'item'):
+            model = self.model(modelType)
+            try:
+                doc = model.load(id, **kwargs)
+                if doc:
+                    doc['_modelType'] = modelType
+                    return doc
+            except ValidationException:
+                pass
+        raise ValidationException(
+            'Invalid document id provided'
+        )
+
+    def iterateItems(self, doc, **kwargs):
+        """Iterate over all items under the given document."""
+        folder = self.model('folder')
+        item = self.model('item')
+        for child in item.find({'folderId': doc['_id']}):
+            yield child
+
+        for child in folder.find({'parentId': doc['_id']}):
+            for subchild in self.iterateItems(child, **kwargs):
+                yield subchild
+
+    def childFolders(self, parent, parentType, cursor=False, filters=None,
+                     limit=0, offset=0, sort=None, **kwargs):
```

Comment:
Is there any virtue to putting the args in the same order as Girder's folder model's `childFolders` method?